js/csv.js

```diff
@@ -0,0 +1,195 @@
+            .map(row => row.trim())
+            .join("\n");
+    }
+
+    // builds the custom pattern that is going to try to
+    // match the delimiter and the escaped and non escaped
+    // values to properly pars the CSV values
+    const pattern = new RegExp(
+        "(\\" +
+            delimiter +
+            "|\\r?\\n|\\r|^)" +
+            '(?:"([^"]*(?:""[^"]*)*)"|' +
+            '([^"\\' +
+            delimiter +
+            "\\r\\n]*))",
+        "gi"
+    );
+
+    // creates the initial data structure already populated
+    // with an initial value as we need to "look-back"
+    const data = [[]];
+
+    // creates an array to hold our individual pattern
+    // matching groups
+    let matches = null;
+
+    // keeps looping over the regular expression matches
+    // until we can no longer find a match
+    while ((matches = pattern.exec(dataS))) {
+        // retrieves the delimiter that was found, this will
+        // affect the newline operation
+        const matchedDelimiter = matches[1];
+
+        // checks to see if the given delimiter has a length
+        // (if it0s not the start of string) and if it matches
+        // field delimiter, if id does not, then we know
+        // that this delimiter is a row delimiter
+        if (matchedDelimiter.length && matchedDelimiter !== delimiter) {
+            // since we have reached a new row of data,
+            // adds an empty row to our data array
+            data.push([]);
+        }
+
+        // allocates space for the value that has been matched
+        let value;
+
+        // now that we have our delimiter out of the way,
+        // checks to see which kind of value we captured
+        // (can be quoted or unquoted)
+        if (matches[2]) {
+            // as a quoted value has been found we need to
+            // unescape the double quotes
+            value = matches[2].replace(/""/g, '"');
+        } else {
+            // a simple (non quoted) value is found and as
+            // such it should be considered the matched value
+            // without any kind of manipulation
+            value = matches[3];
+        }
+
+        // adds the newly (matched) value to the last item in
+        // the data array sequence
+        data[data.length - 1].push(value);
+    }
+
+    if (!object) return data;
+    return _toObject(data);
+};
+
+/**
+ * Converts a sequence of parsed data items into a sequence of
+ * object like items structured according to the CVS.
+ *
+ * @param {Array} data The array of data items defined as sequences
+ * of data items, the first element should be the header.
+ * @returns {Array} An array of object items equivalent to the provided
+ * data array structure provided as input.
+ */
+export const _toObject = data => {
+    const objects = [];
+    const header = data[0];
+    const items = data.slice(1);
+    for (const item of items) {
+        const object = {};
+        objects.push(object);

```

Comment:
I would move this line beneath the for, this is weird.

Reply:
Makes sense, though this was part of the code that already existed in `ripe-util-vue`. The only part that was actually new was the `objectToCsv` function.